sale_order_type/models/sale_order.py

```diff
@@ -35,6 +35,9 @@ def onchange_type_id(self):
             self.payment_term = self.type_id.payment_term_id.id
         if self.type_id.pricelist_id:
             self.pricelist_id = self.type_id.pricelist_id.id
+            res = self.onchange_pricelist_id(
+                self.pricelist_id.id, self.order_line.ids)
+            self.currency_id = res['value']['currency_id']
```

Comment:
You should make `self.update(res.get('value', {}))` for updating all possible values returned from onchange.